Build error java

### Minecolonies version

version 0.8

### Expected behavior

./gradlew build
...
>BUILD SUCCESSFUL

### Actual behaviour

i have error on **"./gradlew build"** and
When start server minecraft , mod isnt launch

### Steps to reproduce the problem

**1/** ./gradlew build
```

This mapping 'snapshot_20161209' was designed for MC 1.11! Use at your own peril.
Cannot run the CurseUpload sequence. No API-Key or release type has been provided.
Found AccessTransformer: minecolonies_at.cfg
:deobfCompileDummyTask
:deobfProvidedDummyTask
:sourceApiJava UP-TO-DATE
:compileApiJava
Note: /home/game/minecraft/minecolonies/build/sources/api/java/com/minecolonies/api/colony/requestsystem/factory/IFactoryController.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:processApiResources UP-TO-DATE
:apiClasses
:apiJar UP-TO-DATE
:sourceBlockOutJava UP-TO-DATE
:compileBlockOutJava UP-TO-DATE
:processBlockOutResources UP-TO-DATE
:blockOutClasses UP-TO-DATE
:blockOutJar UP-TO-DATE
:sourceMainJava UP-TO-DATE
:sourceStructuresJava UP-TO-DATE
:compileStructuresJava UP-TO-DATE
:processStructuresResources UP-TO-DATE
:structuresClasses UP-TO-DATE
:compileJava
**/home/game/minecraft/minecolonies/build/sources/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:861: warning: [unchecked] unchecked cast
            return (T) new InvWrapper(inventory);
                       ^
  required: T
  found:    InvWrapper
  where T is a type-variable:
    T extends Object declared in method <T>getCapability(Capability<T>,EnumFacing)
/home/game/minecraft/minecolonies/build/sources/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java:77: warning: [rawtypes] found raw type: INBTSerializable
                final INBTSerializable serializable = (INBTSerializable) handlerModifiable;
                      ^
  missing type arguments for generic class INBTSerializable<T>
  where T is a type-variable:
    T extends NBTBase declared in interface INBTSerializable
/home/game/minecraft/minecolonies/build/sources/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java:109: warning: [rawtypes] found raw type: INBTSerializable
                    final INBTSerializable serializable = (INBTSerializable) modifiable;
                          ^
  missing type arguments for generic class INBTSerializable<T>
  where T is a type-variable:
    T extends NBTBase declared in interface INBTSerializable
/home/game/minecraft/minecolonies/build/sources/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java:110: warning: [unchecked] unchecked call to deserializeNBT(T) as a member of the raw type INBTSerializable
                    serializable.deserializeNBT(handlerCompound);
                                               ^
  where T is a type-variable:
    T extends NBTBase declared in interface INBTSerializable
4 warnings**
:processResources UP-TO-DATE
:classes
:deobfJar UP-TO-DATE
:structuresJar UP-TO-DATE
:jar
:sourceTestJava UP-TO-DATE
:compileTestJava
:processTestResources UP-TO-DATE
:testClasses
:test
:extractMcpData SKIPPED
:extractMcpMappings SKIPPED
:getVersionJson
:extractUserdev UP-TO-DATE
:genSrgs SKIPPED
:reobfJar
:extractRangemapReplacedMain
/home/game/minecraft/minecolonies/build/sources/main/java
:retromapReplacedMain UP-TO-DATE
:sourceJar UP-TO-DATE
:assemble
:check
:signJar SKIPPED
:build

BUILD SUCCESSFUL

```


_PS : you need to update your README.md_

--- a/build.gradle
+++ b/build.gradle
@@ -149,6 +149,16 @@ compileJava {
     options.compilerArgs << "-Xlint"
 }
 
+compileApiJava {
+    options.encoding = 'UTF-8'
+    options.compilerArgs << "-Xlint"
+}
+
+compileBlockOutJava {
+    options.encoding = 'UTF-8'
+    options.compilerArgs << "-Xlint"
+}
+
 task apiJar(type: Jar) {
     from sourceSets.api.output
     classifier 'api'

--- a/src/api/java/com/minecolonies/api/colony/requestsystem/StandardFactoryController.java
+++ b/src/api/java/com/minecolonies/api/colony/requestsystem/StandardFactoryController.java
@@ -10,7 +10,6 @@ import java.util.HashMap;
 import java.util.Map;
 
 import static com.minecolonies.api.util.constant.Suppression.*;
-import static com.minecolonies.api.util.constant.Suppression.UNCHECKED;
 
 /**
  * Default implementation of a FactoryController

--- a/src/api/java/com/minecolonies/api/colony/requestsystem/StandardFactoryController.java
+++ b/src/api/java/com/minecolonies/api/colony/requestsystem/StandardFactoryController.java
@@ -3,12 +3,14 @@ package com.minecolonies.api.colony.requestsystem;
 import com.minecolonies.api.colony.requestsystem.factory.IFactory;
 import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
 import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.Suppression;
 import net.minecraft.nbt.NBTTagCompound;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.HashMap;
 import java.util.Map;
 
+import static com.minecolonies.api.util.constant.Suppression.*;
 import static com.minecolonies.api.util.constant.Suppression.UNCHECKED;
 
 /**
@@ -30,11 +32,13 @@ public final class StandardFactoryController implements IFactoryController
     /**
      * Input mappings.
      */
+    @SuppressWarnings(RAWTYPES)
     @NotNull
     private final        Map<Class, IFactory>      inputMappings  = new HashMap<>();
     /**
      * Output mappings.
      */
+    @SuppressWarnings(RAWTYPES)
     @NotNull
     private final        Map<Class, IFactory>      outputMappings = new HashMap<>();
 
@@ -88,7 +92,7 @@ public final class StandardFactoryController implements IFactoryController
      * @throws IllegalArgumentException is thrown when the given output class is unknown to this Factory Controller.
      */
     @Override
-    @SuppressWarnings(UNCHECKED)
+    @SuppressWarnings({UNCHECKED, RAWTYPES})
     public <Output> IFactory<?, Output> getFactoryForOutput(@NotNull final Class<? extends Output> clazz) throws IllegalArgumentException
     {
         if (!inputMappings.containsKey(clazz))

--- a/src/api/java/com/minecolonies/api/colony/requestsystem/StandardFactoryController.java
+++ b/src/api/java/com/minecolonies/api/colony/requestsystem/StandardFactoryController.java
@@ -3,7 +3,6 @@ package com.minecolonies.api.colony.requestsystem;
 import com.minecolonies.api.colony.requestsystem.factory.IFactory;
 import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
 import com.minecolonies.api.util.Log;
-import com.minecolonies.api.util.constant.Suppression;
 import net.minecraft.nbt.NBTTagCompound;
 import org.jetbrains.annotations.NotNull;
 

--- a/src/api/java/com/minecolonies/api/colony/requestsystem/factory/IFactoryController.java
+++ b/src/api/java/com/minecolonies/api/colony/requestsystem/factory/IFactoryController.java
@@ -3,6 +3,8 @@ package com.minecolonies.api.colony.requestsystem.factory;
 import net.minecraft.nbt.NBTTagCompound;
 import org.jetbrains.annotations.NotNull;
 
+import static com.minecolonies.api.util.constant.Suppression.UNCHECKED;
+
 /**
  * Interface used to describe classes that function as Factory controllers.
  */
@@ -18,6 +20,7 @@ public interface IFactoryController
      *
      * @throws IllegalArgumentException is thrown when the given input name is unknown to this Factory Controller.
      */
+    @SuppressWarnings(UNCHECKED)
     default <Input> IFactory<Input, ?> getFactoryForInput(@NotNull final String className) throws IllegalArgumentException
     {
         //Simple default implementation grabs the class from a given name and casts it to the proper type.
@@ -56,6 +59,7 @@ public interface IFactoryController
      *
      * @throws IllegalArgumentException is thrown when the given Output name is unknown to this Factory Controller.
      */
+    @SuppressWarnings(UNCHECKED)
     default <Output> IFactory<?, Output> getFactoryForOutput(@NotNull final String className) throws IllegalArgumentException
     {
         //Simple default implementation grabs the class from a given name and casts it to the proper type.

--- a/src/api/java/com/minecolonies/api/colony/requestsystem/token/StandardToken.java
+++ b/src/api/java/com/minecolonies/api/colony/requestsystem/token/StandardToken.java
@@ -5,6 +5,8 @@ import org.jetbrains.annotations.NotNull;
 
 import java.util.UUID;
 
+import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;
+
 /**
  * Internal implementation of the IToken interface.
  * Uses UUID to store the ID of the request.
@@ -71,6 +73,7 @@ public class StandardToken implements IToken<UUID, NBTTagCompound>
     }
 
     @Override
+    @SuppressWarnings(RAWTYPES)
     public boolean equals(final Object o)
     {
         if (this == o)

--- a/src/api/java/com/minecolonies/api/util/constant/Suppression.java
+++ b/src/api/java/com/minecolonies/api/util/constant/Suppression.java
@@ -17,6 +17,14 @@ public final class Suppression
     public static final String UNCHECKED = "unchecked";
 
     /**
+     * Suppress warnings for raw type annotation.
+     * <p>
+     * We sometimes need this for complicated typings.
+     */
+    @NonNls
+    public static final String RAWTYPES = "rawtypes";
+
+    /**
      * Suppress warnings for deprecations.
      * <p>
      * We sometimes need this for minecraft methods we have to keep support for.
@@ -87,6 +95,7 @@ public final class Suppression
      */
     @NonNls
     public static final String RESOURCES_SHOULD_BE_CLOSED = "squid:S2095 ";
+
 
     private Suppression()
     {

--- a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java
+++ b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java
@@ -61,6 +61,7 @@ import java.lang.reflect.Field;
 import java.util.*;
 
 import static com.minecolonies.api.util.constant.Suppression.INCREMENT_AND_DECREMENT_OPERATORS_SHOULD_NOT_BE_USED_IN_A_METHOD_CALL_OR_MIXED_WITH_OTHER_OPERATORS_IN_AN_EXPRESSION;
+import static com.minecolonies.api.util.constant.Suppression.UNCHECKED;
 
 /**
  * The Class used to represent the citizen entities.
@@ -853,6 +854,7 @@ public class EntityCitizen extends EntityAgeable implements INpc
         return colony;
     }
 
+    @SuppressWarnings(UNCHECKED)
     @Override
     public <T> T getCapability(final Capability<T> capability, final EnumFacing facing)
     {

--- a/src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java
+++ b/src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java
@@ -16,6 +16,9 @@ import org.jetbrains.annotations.Nullable;
 
 import java.util.Arrays;
 
+import static com.minecolonies.api.util.constant.Suppression.RAWTYPES;
+import static com.minecolonies.api.util.constant.Suppression.UNCHECKED;
+
 /**
  * Abstract class wrapping around multiple IItemHandler.
  */
@@ -62,6 +65,7 @@ public abstract class AbstractCombinedItemHandler
         this.defaultName = defaultName;
     }
 
+    @SuppressWarnings(RAWTYPES)
     @Override
     public NBTTagCompound serializeNBT()
     {
@@ -92,6 +96,7 @@ public abstract class AbstractCombinedItemHandler
         return compound;
     }
 
+    @SuppressWarnings({RAWTYPES, UNCHECKED})
     @Override
     public void deserializeNBT(final NBTTagCompound nbt)
     {

